cv-data-service-library/src/test/java/com/trihydro/library/helpers/UtilityTest.java

```diff
@@ -254,21 +238,87 @@ public void calculateAnchorCoordinateExample4() throws Utility.IdenticalPointsEx
 
         // Act & Assert
         Assertions.assertThrows(Utility.IdenticalPointsException.class, () -> {
             uut.calculateAnchorCoordinate(firstMilepost, secondMilepost);
         });
     }
 
+    @Test
+    public void calculateAnchorCoordinate_identicalPoints_positiveLat_negativeLon() {
+        // Arrange
+        BigDecimal lat = BigDecimal.valueOf(40);
+        BigDecimal lon = BigDecimal.valueOf(-100);
+        Milepost firstMilepost = createMilepost(lat, lon);
+        Milepost secondMilepost = createMilepost(lat, lon);
+
+        // Act & Assert
+        Assertions.assertThrows(Utility.IdenticalPointsException.class, () -> {
+            uut.calculateAnchorCoordinate(firstMilepost, secondMilepost);
+        });
+    }
+
+    @Test
+    public void calculateAnchorCoordinate_identicalPoints_negativeLat_positiveLon() {
+        // Arrange
+        BigDecimal lat = BigDecimal.valueOf(-40);
+        BigDecimal lon = BigDecimal.valueOf(100);
+        Milepost firstMilepost = createMilepost(lat, lon);
+        Milepost secondMilepost = createMilepost(lat, lon);
+
+        // Act & Assert
+        Assertions.assertThrows(Utility.IdenticalPointsException.class, () -> {
+            uut.calculateAnchorCoordinate(firstMilepost, secondMilepost);
+        });
+    }
+
+    @Test
+    public void calculateAnchorCoordinate_identicalPoints_positiveLat_positiveLon() {
+        // Arrange
+        BigDecimal lat = BigDecimal.valueOf(-40);
+        BigDecimal lon = BigDecimal.valueOf(100);
+        Milepost firstMilepost = createMilepost(lat, lon);
+        Milepost secondMilepost = createMilepost(lat, lon);
+
+        // Act & Assert
+        Assertions.assertThrows(Utility.IdenticalPointsException.class, () -> {
+            uut.calculateAnchorCoordinate(firstMilepost, secondMilepost);
+        });
+    }
+
+    @Test
+    public void calculateAnchorCoordinate_identicalPoints_negativeLat_negativeLon() {
+        // Arrange
+        BigDecimal lat = BigDecimal.valueOf(-40);
+        BigDecimal lon = BigDecimal.valueOf(-100);
+        Milepost firstMilepost = createMilepost(lat, lon);
+        Milepost secondMilepost = createMilepost(lat, lon);
+
+        // Act & Assert
+        Assertions.assertThrows(Utility.IdenticalPointsException.class, () -> {
+            uut.calculateAnchorCoordinate(firstMilepost, secondMilepost);
+        });
+    }
+
+    /**
+     * Creates a new Milepost instance and sets its latitude and longitude values.
+     *
+     * @param latitude  the latitude of the Milepost
+     * @param longitude the longitude of the Milepost
+     * @return a newly created Milepost instance with the specified latitude and longitude
+     */
+    private Milepost createMilepost(BigDecimal latitude, BigDecimal longitude) {
+        Milepost milepost = new Milepost();
+        milepost.setLatitude(latitude);
+        milepost.setLongitude(longitude);
+        return milepost;
+    }
```

Comment:
Suggestion (non-blocking): This method could be replaced with a new constructor for the *Milepost* class. Since *Milepost* objects are instantiated elsewhere, using a constructor would be a cleaner and more consistent approach than a private `createMilepost` method.

Reply:
Addressed